api/lib/domain/usecases/delete-attachment.js

```diff
@@ -0,0 +1,32 @@
+export async function deleteAttachment({
+  attachmentId,
+  attachmentRepository,
+  challengeRepository,
+  createChallengeTransformer,
+  updatedRecordNotifier,
+  pixApiClient,
+  logger,
+  Sentry,
+}) {
+  const attachmentToDelete = await attachmentRepository.get(attachmentId);
```

Comment:
Il faudrait mettre une guarde après ce `get` pour vérifier que l'attachment existe bien avant de continuer.
```suggestion
  const attachmentToDelete = await attachmentRepository.get(attachmentId);
  if (!attachmentToDelete) throw NotFoundError(...);
```